test/NuGet.Core.Tests/NuGet.PackageManagement.Test/ResolverGatherTests.cs

```diff
@@ -1420,6 +1421,165 @@ public async Task ResolverGather_DependenciesSpreadAcrossRepos()
+        [InlineData("public,nuget", "nuget")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.MVC")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.MVC.ASP")]
+        [InlineData("public,Contoso.Opensource.* ", "Contoso.Opensource.MVC.ASP")]
+        [InlineData("nuget.org,nuget|privateRepository,private*", "nuget")]
+        [InlineData("nuget.org,nuget|privateRepository,private*", "private1")]
+        public async Task ResolverGather_PackageNamespace_Succeed(string packageNamespaces, string packageId)
+        {
+            // Arrange
+            PackageNamespacesConfiguration namespacesConfiguration = PackageNamespacesConfigurationUtility.GetPackageNamespacesConfiguration(packageNamespaces);
+            IReadOnlyList<string> configuredSources = namespacesConfiguration.GetConfiguredPackageSources(packageId);
+            var target = new PackageIdentity(packageId, new NuGetVersion(1, 0, 0));
+            IEnumerable<PackageIdentity> targets = new[] { target };
+
+            var framework = NuGetFramework.Parse("net451");
+
+            var packages1 = new List<SourcePackageDependencyInfo>
+                {
+                    new SourcePackageDependencyInfo(packageId, new NuGetVersion(1, 0, 0), new Packaging.Core.PackageDependency[] { }, true, null)
+                };
+
+            var packages2 = new List<SourcePackageDependencyInfo>
+                {
+                    new SourcePackageDependencyInfo(packageId, new NuGetVersion(1, 0, 0),  new Packaging.Core.PackageDependency[] { }, true, null)
+                };
+
+            var packages3 = new List<SourcePackageDependencyInfo>
+                {
+                    new SourcePackageDependencyInfo(packageId, new NuGetVersion(1, 0, 0),  new Packaging.Core.PackageDependency[] { }, true, null)
+                };
+
+
+            var providers1 = new List<Lazy<INuGetResourceProvider>>();
+            providers1.Add(new Lazy<INuGetResourceProvider>(() => new TestDependencyInfoProvider(packages1)));
+
+            var providers2 = new List<Lazy<INuGetResourceProvider>>();
+            providers2.Add(new Lazy<INuGetResourceProvider>(() => new TestDependencyInfoProvider(packages2)));
+
+            var providers3 = new List<Lazy<INuGetResourceProvider>>();
+            providers3.Add(new Lazy<INuGetResourceProvider>(() => new TestDependencyInfoProvider(packages3)));
+
+            var repos = new List<SourceRepository>();
+            repos.Add(new SourceRepository(new Configuration.PackageSource("http://1", "someRepository1"), providers1));
+            repos.Add(new SourceRepository(new Configuration.PackageSource("http://2", "someRepository2"), providers2));
+            repos.Add(new SourceRepository(new Configuration.PackageSource("http://3", configuredSources[0]), providers3));
+
+            var context = new GatherContext(namespacesConfiguration);
```

Comment:
nit: If you want to improve readability consider switching to something like:

```cs
            var context = new GatherContext(namespacesConfiguration) 
            {
               PrimaryTargets = targets.ToList();
            } 

```

The quick actions are likely to suggest this.

Reply:
Ok. Done.